### PR TITLE
feat: Replace ad hoc auth header with internet standard bearer token auth header

### PR DESCRIPTION
**Pull request by SleeplessOne1917**
Does as the title says

The only file that was changed that wasn't a Cargo.toml/lock file was the session middleware. All the other changes were from running `cargo fmt` to try to satisfy the woodpecker build to no avail.

### CODE REVIEW

File: docker/federation/docker-compose.yml [hunk 1]
```diff
@@ -2,11 +2,11 @@ version: "3.7"
...
   init: true
-  image: dessalines/lemmy-ui:0.18.4
+  # image: dessalines/lemmy-ui:0.18.4
   # assuming lemmy-ui is cloned besides lemmy directory
-  # build:
-  #   context: ../../../lemmy-ui
-  #   dockerfile: dev.dockerfile
+  build:
+    context: ../../../lemmy-ui
+    dockerfile: dev.dockerfile
```
Comment (Nutomic) [inline docker/federation/docker-compose.yml:9]: Revert this.

File: crates/api_crud/Cargo.toml [hunk 2]
```diff
@@ -18,6 +18,7 @@ activitypub_federation = { workspace = true }
 bcrypt = { workspace = true }
 serde = { workspace = true }
 actix-web = { workspace = true }
+actix-web-httpauth = { workspace = true }
```
Comment (Nutomic) [inline crates/api_crud/Cargo.toml:21]: This is unused.

**Comment by Nutomic** [comment]
I already implemented this in https://github.com/LemmyNet/lemmy/pull/3818, but your change is much cleaner so lets use that. 

To make the API tests pass you have to set the correct header there, analogous to this change from my PR: https://github.com/LemmyNet/lemmy/pull/3818/commits/b2bacd1a30f92ea3456a2af08d872bb991e9dfa7

**Review by dessalines — APPROVED** [review]
Sweet, thx.

**Pull request merged**